en/docs/mi-analytics/mi-custom-analytics.md

```diff
@@ -0,0 +1,106 @@
+
+This section covers how to configure the `pom.xml` file, implement the required classes, and build the sample project.
+
+### Configuring the pom.xml
+
+Add the WSO2 Nexus repository to the `pom.xml` file:
+
+```xml
+    <repository>
+        <id>wso2-nexus</id>
+        <name>WSO2 internal Repository</name>
+        <url>https://maven.wso2.org/nexus/content/groups/wso2-public/</url>
+        <releases>
+            <enabled>true</enabled>
+            <updatePolicy>daily</updatePolicy>
+            <checksumPolicy>ignore</checksumPolicy>
+        </releases>
+    </repository>
+```
+
+Add the following dependencies.
+
+```xml
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.ei</groupId>
+            <artifactId>org.wso2.micro.integrator.analytics.messageflow.data.publisher</artifactId>
+            <version>${wso2.mi.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.synapse</groupId>
+            <artifactId>synapse-extensions</artifactId>
+            <version>${synapse.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.synapse</groupId>
+            <artifactId>synapse-core</artifactId>
+            <version>${synapse.version}</version>
+        </dependency>
+    </dependencies>
+```
+
+!!! info
+    The versions for ${wso2.mi.version} and ${synapse.version} can be found in the jar versions available in the current Micro Integrator package.
+
+### Implementing required class
+
+To add custom analytics data to the existing event schema, implement a class from the `org.wso2.micro.integrator.analytics.messageflow.data.publisher.producer.AnalyticsCustomDataProvider` interface. Override the `getCustomProperties` method to push custom analytics data.
+
+An example implementation of this class is shown below:
+
+``` java
+    public class SampleCustomDataProvider implements AnalyticsCustomDataProvider {
+        @Override
+        public Map<String, Object> getCustomProperties(MessageContext messageContext) {
+            Map<String, Object> properties = new HashMap<>();
+            Mediator mediator = messageContext.getMainSequence();
+            properties.put("mediatorName", mediator.getMediatorName());
+            return properties;
+        }
+    }
+```
+
+### Build the project
+
+To build the project, run the following command:
+
+```
+    mvn clean install
+```
+
+## Configuring the sample
+
+This section outlines the steps required to configure WSO2 Micro Integrator for the sample created above.
+
+After building the project, copy the created .jar file into the `wso2mi-{version}/lib` directory.
+
+Edit the `analytics` configuration in the `deployment.toml` file, located inside the `wso2mi-{version}/conf` directory, and add the following configuration:
```

Comment:
We usually follow the convention of using the `<MI_HOME>` placeholder instead of `wso2mi-{version}`.

Reply:
Fixed with e2e4aaf143e54d7ca0242e669adee788bfff5196